services/QuillLMS/client/app/bundles/Staff/components/evidence/llmPromptDatasets/trialRunner/promptExamplesSection.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import moment from 'moment';
+
+import NewPromptExampleModal from './newPromptExampleModal'
+
+import { getPromptExamplesForDataset, } from '../../../../utils/evidence/genAIAPIs';
+import { DataTable, } from '../../../../../Shared/index';
+import { PromptExampleInterface, } from '../../../../interfaces/evidenceInterfaces';
+import { OPTIMAL, SUBOPTIMAL, } from '../../../../../../constants/evidence'
+
+function getRandomSample(array) {
+  // Get a random number of items to sample, between 1 and the length of the array
+  const sampleSize = Math.floor(Math.random() * array.length) + 1;
+
+  // Shuffle the array and slice it to get the sample
+  const shuffledArray = array.sort(() => Math.random() - 0.5);
+  return shuffledArray.slice(0, sampleSize);
+}
```

Comment:
it might be worth moving this to the Shared bundle in case we want to use it in the future elsewhere?

Reply:
I think this makes sense as a call-out, but I prefer not to abstract stuff until we have a second use case for it.